grpc/src/main/java/io/micrometer/core/instrument/binder/grpc/TaggingMetricCollectingServerInterceptor.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.micrometer.core.instrument.binder.grpc;
+
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.Status;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import io.stargate.grpc.metrics.api.GrpcMetricsTagProvider;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+
+/**
+ * Extension of the micrometer's {@link MetricCollectingServerInterceptor} that enables us to add
+ * extra tags on each {@link ServerCall}.
+ */
+public class TaggingMetricCollectingServerInterceptor extends MetricCollectingServerInterceptor {
+
+  /**
+   * The total number of requests received, same as
+   * MetricCollectingServerInterceptor#METRIC_NAME_SERVER_REQUESTS_RECEIVED.
+   */
+  private static final String METRIC_NAME_SERVER_REQUESTS_RECEIVED =
```

Comment:
Unfortunate that `MetricCollectingServerInterceptor` made these constants private... or it would have been nice if `newRequestCounterFor()` etc. returned builders.

Reply:
Yes this is really unfortunate.. Maybe I can try to provide a fix for the micrometer quickly..